app/models/course/condition/achievement.rb

```diff
@@ -10,6 +10,11 @@ class Course::Condition::Achievement < ActiveRecord::Base
   delegate :title, to: :achievement
   alias_method :dependent_object, :achievement
 
+  # Trigger for resolving the conditional for a course user
```

Comment:
This should be together with the rest of the validations for Course::Condition::Achievement